OutSystems/outsystems-ui
# ROU-4676: DatePicker issue when setting MinDate

This PR is for fixing an issue on Datepicker to update values when datepicker's input is empty


### What was happening

-
![image](https://github.com/OutSystems/outsystems-ui/assets/25321845/d7ba04ae-aba3-41e6-9bc3-049481832914)


### What was done

- Removed the validation on changeProperty when the InitialDate is updated
- Added the support to OnClose event and add the trigger of the selected dates when the datepicker is closed
- Add support to custom callback that can be defined through platform to attach custom code when the datepicker close event occurs
![2024-01-12 11 50 50](https://github.com/OutSystems/outsystems-ui/assets/25321845/0a3aee69-fed6-43c0-b86e-34f6e0deffb6)


### Test Steps

1. Open Sample page
2. Select a date on first datepicker (this will affect the nested datepicker with the date selected)
3. Select a date on second datepicker
4. Remove the date from the second datepicker's input and close the datepicker by clicking on body
5. Remove the date from the first datepicker's input and close the datepicker by clicking on body
6. Expected: The datepicker will not have any date added to input

### Screenshots
![2024-01-12 11 50 50](https://github.com/OutSystems/outsystems-ui/assets/25321845/5b0b8146-0ff0-4c2a-9c9b-323f544ffbfc)

### Checklist
